modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -2027,20 +2027,43 @@ private void _updateDescendantNodeObjectDefinitions(
 
 		deployObjectDefinition(objectDefinition1);
 
+		ObjectDefinitionTreeFactory objectDefinitionTreeFactory =
+			new ObjectDefinitionTreeFactory(
+				objectDefinitionLocalService, _objectRelationshipLocalService);
+
 		for (ObjectRelationship objectRelationship : objectRelationships) {
 			ObjectDefinition objectDefinition2 =
 				objectDefinitionPersistence.findByPrimaryKey(
 					objectRelationship.getObjectDefinitionId2());
 
 			if (!objectDefinition2.isApproved()) {
+				Tree tree = objectDefinitionTreeFactory.create(
+					false, objectDefinition1.getObjectDefinitionId());
+
+				Node rootNode = tree.getRootNode();
+
+				for (Node childNode : rootNode.getChildNodes()) {
+					Iterator<Node> iterator = tree.iterator(
+						childNode.getPrimaryKey());
+
+					while (iterator.hasNext()) {
+						Node node = iterator.next();
+
+						ObjectDefinition nodeObjectDefinition =
+							objectDefinitionLocalService.getObjectDefinition(
+								node.getPrimaryKey());
+
+						nodeObjectDefinition.setRootObjectDefinitionId(
+							childNode.getPrimaryKey());
+
+						objectDefinitionPersistence.update(
+							nodeObjectDefinition);
+					}
+				}
```

Comment:
Let's discuss this in person, I think it is easier :+1:

Reply:
ok